### PR TITLE
Disable autocompletion for authorization secrets

### LINKED ISSUE #9921: Disable autocompletion for authorization secrets
When inputting values for the API key and bearer token, there's a possibility of autocompletion, which can potentially reveal previously used secrets. 
We want these values to remain visible to users as they type, while ensuring that they are not exposed through autocompletion, similar to how it happens when creating secrets in GitHub repositories. As such, we propose to implement a simple solution by setting `autoComplete=off` for these inputs:
https://github.com/swagger-api/swagger-ui/blob/1367a8fbdfddd697b8c71493bb09c01baf17d5a3/src/core/components/auth/api-key-auth.jsx#L73-L78

https://github.com/swagger-api/swagger-ui/blob/1367a8fbdfddd697b8c71493bb09c01baf17d5a3/src/core/plugins/oas3/components/auth/http-auth.jsx#L132-L138

### Additional context or thoughts
This issue was raised with https://github.com/swagger-api/swagger-ui/pull/9858

### PR DESCRIPTION
**Pull request by klm-turing**
### Description
This pull request set `autoComplete=off` to the inputs below so as not to expose the bearer's API key and token by autocompletion when typed.

https://github.com/swagger-api/swagger-ui/blob/1367a8fbdfddd697b8c71493bb09c01baf17d5a3/src/core/components/auth/api-key-auth.jsx#L73-L78

https://github.com/swagger-api/swagger-ui/blob/1367a8fbdfddd697b8c71493bb09c01baf17d5a3/src/core/plugins/oas3/components/auth/http-auth.jsx#L132-L138

### Motivation and Context
close #9921 

### How Has This Been Tested?
I opened my browser with `localhost:3200` . I clicked on authorize, start typing a secret and closed the modal window.
I reopened the modal window, logged out and started typing a secret again. Autocomplete did not appear.

### Screenshots (if appropriate):
[Screencast from 2024-05-13 20-47-19.webm](https://github.com/swagger-api/swagger-ui/assets/167549638/dc0c52df-9abf-4e71-b885-7b316c8a198d)

### My PR contains... 

- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [ ] Bug fixes (non-breaking change which fixes an issue)
- [x] Improvements (misc. changes to existing features)
- [x] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [x] are not breaking changes.

### Documentation
- [x] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [x] My changes can not or do not need to be tested.
- [ ] My changes can and should be tested by unit and/or integration tests.
- [ ] If yes to above: I have added tests to cover my changes.
- [ ] If yes to above: I have taken care to cover edge cases in my tests.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by nejch** [comment]
Since this is the replacement for https://github.com/swagger-api/swagger-ui/pull/9858 which was closed a while back, does this PR have a better chance of getting merged? :bow:

File: src/core/plugins/oas3/components/auth/http-auth.jsx [hunk 1]
```diff
@@ -82,6 +82,7 @@ export default class HttpAuth extends React.Component {
                     name="username"
                     aria-label="auth-basic-username"
                     onChange={ this.onChange }
+                    autoComplete='off'
```
Comment (glowcloud) [inline src/core/plugins/oas3/components/auth/http-auth.jsx]: We don't need to disable autocompletion for usernames. I believe this one was meant to disable bearer input: https://github.com/swagger-api/swagger-ui/blob/85f0c5f8c3da20c6a5b58579248d5b256dcd83e1/src/core/plugins/oas3/components/auth/http-auth.jsx#L134-L140

**Review by glowcloud — COMMENTED** [review]
Hi @klm-turing,

Thanks for your contribution! We're revisiting some older PRs and if you're still interested in continuing your work on this one, please take a look at the comments below.
